newrelic/api/error_trace.py

```diff
@@ -13,25 +13,38 @@
 # limitations under the License.
 
 import functools
+import warnings
 
-from newrelic.api.time_trace import current_trace, record_exception
+from newrelic.api.time_trace import current_trace, notice_error
 from newrelic.common.object_wrapper import FunctionWrapper, wrap_object
 
 
 class ErrorTrace(object):
-
-    def __init__(self, ignore_errors=[], **kwargs):
-        parent = None
-        if kwargs:
-            if len(kwargs) > 1:
-                raise TypeError("Invalid keyword arguments:", kwargs)
-            parent = kwargs['parent']
-
+    def __init__(
+        self,
+        ignore_errors=[],
+        ignore=None,
+        expected=None,
+        status_code=None,
+        parent=None,
+    ):
         if parent is None:
             parent = current_trace()
 
         self._transaction = parent and parent.transaction
         self._ignore_errors = ignore_errors
+        self._ignore = ignore if ignore is not None else ignore_errors
+        self._expected = expected
+        self._status_code = status_code
+
+        if self._ignore_errors:
```

Comment:
Can change this to a `if ignore_errors` and drop the line above where we store it since this is never accessed again.